### PR TITLE
Add `xod/json` library

### PR DESCRIPTION
**Pull request by evgenykochetkov**

### CODE REVIEW

File: workspace/__lib__/xod/json/parse-json/patch.xodp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "description": "Pasres a stream of characters that represent a JSON document.",
```
Comment (knopki) [inline workspace/__lib__/xod/json/parse-json/patch.xodp]: Can't pasre this :)

File: workspace/__lib__/xod/json/string-value/patch.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+    auto state = getState(ctx);
+    auto parsed = getValue<input_IN1>(ctx);
+
+    if (isSettingUp()) {
+        // save initial cap to ignore possiple input changes during program execution
```
Comment (knopki) [inline workspace/__lib__/xod/json/string-value/patch.cpp]: inpossiplu

File: workspace/__lib__/xod/json/parse-json/patch.cpp [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+#define STACK_OBJECT             0
+#define STACK_ARRAY              1
+#define STACK_KEY                2
+#define STACK_STRING             3
+
+bool isFirstCharInNumber(char c) {
+    return (c >= '0' && c <= '9') || c == '-';
+}
+
+// We don't use `strchr` because buffer we search in is not \0 - terminated
```
Comment (nkrkv) [inline workspace/__lib__/xod/json/parse-json/patch.cpp]: It’s [`memchr`](http://www.cplusplus.com/reference/cstring/memchr/)

File: workspace/__lib__/xod/json/parse-json/patch.cpp [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+        return true;
+      }
+    }
+    return false;
+}
+
+struct ContextObject;
+
+// we only use buffer to validate numbers, booleans and `null`
+#define BUFFER_MAX_LENGTH 64
```
Comment (nkrkv) [inline workspace/__lib__/xod/json/parse-json/patch.cpp]: In modern C++ prefer constexpr

```suggestion
constexpr size_t BUFFER_MAX_LENGTH = 64;
```

And are we really need to keep 64 bytes? The number we can precisely keep is ~7 digits. A 15-digit number is huge. I think we could limit ourselves to 16 chars.

File: workspace/__lib__/xod/json/parse-json/patch.cpp [hunk 5]
```diff
@@ -0,0 +1,446 @@
+
+using xod::xod__json__parsed_json_stream::ParserState;
+
+#define STACK_OBJECT             0
+#define STACK_ARRAY              1
+#define STACK_KEY                2
+#define STACK_STRING             3
```
Comment (nkrkv) [inline workspace/__lib__/xod/json/parse-json/patch.cpp]: Looks like an enum. Let it be enum

File: workspace/__lib__/xod/json/parse-json/patch.cpp [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+
+struct ContextObject;
+
+// we only use buffer to validate numbers, booleans and `null`
+#define BUFFER_MAX_LENGTH 64
+
+class JsonParser {
+  private:
+    ParserState state;
+    int stack[20];
```
Comment (nkrkv) [inline workspace/__lib__/xod/json/parse-json/patch.cpp]: Memory waste? Might be `uint8_t` array.

File: workspace/__lib__/xod/json/parse-json/patch.cpp [hunk 7]
```diff
@@ -0,0 +1,438 @@
+
+using xod::json_parser::ParserState;
+
+enum class Stack : uint8_t {
```
Comment (nkrkv) [inline workspace/__lib__/xod/json/parse-json/patch.cpp]: `StackItem` would be a more correct name

**Pull request merged**